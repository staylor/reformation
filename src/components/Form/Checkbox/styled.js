import { css } from 'pretty-lights';
import themeUtils from 'styles/theme';

// eslint-disable-next-line
export const checkboxClass = css`
  appearance: none;
  background-color: ${themeUtils.colors.white};
  border: 1px solid ${themeUtils.colors.form.checkbox.border};
  box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.1);
  color: ${themeUtils.colors.text};
  cursor: pointer;
  display: inline-block;
  height: 16px;
  line-height: 0;
  min-width: 16px;
  outline: 0;
  padding: 0;
  transition: 0.05s border-color ease-in-out;
  vertical-align: text-top;
  width: 16px;

  &:checked {
    &::before {
      color: ${themeUtils.colors.pink};
      content: '\f147';
      float: left;
      display: inline-block;
      /* stylelint-disable-next-line font-family-no-missing-generic-family-keyword */
      font: normal 21px/1 dashicons;
      margin: -3px 0 0 -4px;
      speak: none;
      vertical-align: middle;
      width: 16px;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
    }
  }
`;
