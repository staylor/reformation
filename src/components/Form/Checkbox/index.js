import React, { Component } from 'react';
import { cx } from 'pretty-lights';
import { checkboxClass } from './styled';

/* eslint-disable react/prop-types */

// type Props = {
//   id: string,
//   className?: string | null,
//   checked?: boolean,
//   onChange?: (checked: boolean, id?: string | null) => void,
//   bindRef?: (element: any) => void,
// };
//
// type State = {
//   checked: boolean,
// };

export default class Checkbox extends Component {
  state = {
    checked: false,
  };

  static getDerivedStateFromProps(nextProps) {
    if (Object.keys(nextProps).includes('checked')) {
      return { checked: Boolean(nextProps.checked) };
    }
    return null;
  }

  onChange = e => {
    const value = Boolean(e.target.checked);
    this.setState({ checked: value }, () => {
      if (this.props.onChange) {
        this.props.onChange(value, this.props.id || null);
      }
    });
  };

  render() {
    const { id, className, checked, onChange, bindRef, ...props } = this.props;

    return (
      <input
        // eslint-disable-next-line
        {...props}
        ref={bindRef}
        className={cx(checkboxClass, className)}
        type="checkbox"
        onChange={this.onChange}
        checked={this.state.checked}
      />
    );
  }
}

Checkbox.defaultProps = {
  className: undefined,
  checked: false,
  onChange: () => null,
  bindRef: () => null,
};
