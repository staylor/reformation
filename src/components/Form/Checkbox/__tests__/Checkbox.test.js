import React from 'react';
import Checkbox from 'components/Form/Checkbox';
import { shallow } from 'enzyme';

describe('Checkbox', () => {
  test('empty', () => {
    const wrapper = shallow(<Checkbox />);

    expect(wrapper).toMatchSnapshot();
  });

  test('checked', () => {
    const wrapper = shallow(<Checkbox checked />);

    expect(wrapper).toMatchSnapshot();
  });

  test('add className', () => {
    const wrapper = shallow(<Checkbox className="foo" />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('onChange', () => {
    // test('toggle', () => {
    //   const func = jest.fn();
    //   const wrapper = shallow(<Checkbox onChange={func} />);
    //   expect(func).not.toHaveBeenCalled();
    //   expect(wrapper.state('checked')).toBe(false);
    //
    //   wrapper.find('input').simulate('change', { target: { checked: true } });
    //   expect(func).toHaveBeenCalled();
    //   expect(wrapper.state('checked')).toBe(true);
    // });
    // test('bad event', () => {
    //   const func = jest.fn();
    //   const wrapper = shallow(<Checkbox checked onChange={func} />);
    //   wrapper.find('input').simulate('change', { target: { value: 'foo' } });
    //   expect(func).toHaveBeenCalled();
    //   expect(wrapper.state('checked')).toBe(false);
    // });
    // test('checked false', () => {
    //   const func = jest.fn();
    //   const wrapper = shallow(<Checkbox onChange={func} />);
    //   wrapper.find('input').simulate('change', { target: { checked: false } });
    //   expect(func).toHaveBeenCalled();
    //   expect(wrapper.state('checked')).toBe(false);
    // });
    //
    // test('toggle false', () => {
    //   const func = jest.fn();
    //   const wrapper = shallow(<Checkbox onChange={func} checked />);
    //   wrapper.find('input').simulate('change', { target: { checked: false } });
    //   expect(func).toHaveBeenCalled();
    //   expect(wrapper.state('checked')).toBe(false);
    // });
    //
    // test('id', () => {
    //   const func = jest.fn();
    //   const wrapper = shallow(<Checkbox id="pizza" onChange={func} checked />);
    //   wrapper.find('input').simulate('change', { target: { checked: false } });
    //   expect(func).toHaveBeenCalledWith(false, 'pizza');
    //   expect(wrapper.state('checked')).toBe(false);
    // });
  });
});
