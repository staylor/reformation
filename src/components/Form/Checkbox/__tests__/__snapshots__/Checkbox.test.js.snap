// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox add className 1`] = `
.lights-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #fff;
  border: 1px solid #b4b9be;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.1);
  color: #999;
  cursor: pointer;
  display: inline-block;
  height: 16px;
  line-height: 0;
  min-width: 16px;
  outline: 0;
  padding: 0;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  vertical-align: text-top;
  width: 16px;
}

.lights-0:checked::before {
  color: #e50082;
  content: '\\f147';
  float: left;
  display: inline-block;
  font: normal 21px/1 dashicons;
  margin: -3px 0 0 -4px;
  speak: none;
  vertical-align: middle;
  width: 16px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

<input
  checked={false}
  className="lights-0 foo"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Checkbox checked 1`] = `
.lights-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #fff;
  border: 1px solid #b4b9be;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.1);
  color: #999;
  cursor: pointer;
  display: inline-block;
  height: 16px;
  line-height: 0;
  min-width: 16px;
  outline: 0;
  padding: 0;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  vertical-align: text-top;
  width: 16px;
}

.lights-0:checked::before {
  color: #e50082;
  content: '\\f147';
  float: left;
  display: inline-block;
  font: normal 21px/1 dashicons;
  margin: -3px 0 0 -4px;
  speak: none;
  vertical-align: middle;
  width: 16px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

<input
  checked={true}
  className="lights-0"
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Checkbox empty 1`] = `
.lights-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #fff;
  border: 1px solid #b4b9be;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.1);
  color: #999;
  cursor: pointer;
  display: inline-block;
  height: 16px;
  line-height: 0;
  min-width: 16px;
  outline: 0;
  padding: 0;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  vertical-align: text-top;
  width: 16px;
}

.lights-0:checked::before {
  color: #e50082;
  content: '\\f147';
  float: left;
  display: inline-block;
  font: normal 21px/1 dashicons;
  margin: -3px 0 0 -4px;
  speak: none;
  vertical-align: middle;
  width: 16px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

<input
  checked={false}
  className="lights-0"
  onChange={[Function]}
  type="checkbox"
/>
`;
